projects/storefrontlib/src/cms-components/product/product-variant-selector/size-selector/size-selector.component.ts

```diff
@@ -0,0 +1,37 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import {
+  OccConfig,
+  Product,
+  RoutingService,
+  BaseOption,
+} from '@spartacus/core';
+
+@Component({
+  selector: 'cx-size-selector',
+  templateUrl: './size-selector.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class VariantSizeSelectorComponent {
+  constructor(
+    private config: OccConfig,
+    private routingService: RoutingService
+  ) {}
+
+  @Input()
+  product: Product;
+
+  @Input()
+  variants: BaseOption;
+
+  baseUrl = this.config.backend.occ.baseUrl;
```

Comment:
Do we need it as a public property?
Also, for config, it's safer to use getters.